internal/api/http/middleware/database.go

```diff
@@ -30,19 +26,25 @@ func (w *ResponseWriterWrapper) StatusCode() int {
 func DatabaseMiddleware(next http.Handler, db database.Database) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
-		tx := db.Connect().Begin()
-		if tx.Error != nil {
+		tx, err := db.Connect()
```

Comment:
Scenario: in a router you start a new transaction which overwrites the db.tx to the current transaction than in the service an error occurs and the tx.Rollback is used which will set the shouldRollback to true. This means that after the router has finished, the execution will go into the if statement on line 38 which will rollback the db.tx which in this case is the transaction started in the service meaning that the transaction opened in this middleware will never be closed committed or rollbacked

Reply:
This scenario is not possible due to how db.Connect() function is [implemented](https://github.com/mini-maxit/backend/pull/9/commits/21d09a5d8224f0a74223263cd38283398f180df9#diff-c7b558b1e970d3c95a1a25710358890827b51731ca21577a67dd5cd6049f9685R28-R39).
You can see that if the transaction is already started than it is returned. This is done exactly to prevent multuple opening transactions

Reply:
ahhhh good did not see that